### PR TITLE
Fix param_scheduler trigger events in doc examples

### LINKED ISSUE #425: Wrong event used in doc example of "ignite.contrib.handlers.LRScheduler"
At first, thanks for this nice project! 

While reading the documentation, I encountered, that in the  little example on how to use `ignite.contrib.handlers.LRScheduler` the `ITERATION_COMPLETED` event is used instead of the `ITERATION_STARTED` event.  From pytorch documentation, the event should be triggered, when the iteration started.

Example from pytorch documentation ([doc link](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.StepLR)):

```python
>>> # Assuming optimizer uses lr = 0.05 for all groups
>>> # lr = 0.05     if epoch < 30
>>> # lr = 0.005    if 30 <= epoch < 60
>>> # lr = 0.0005   if 60 <= epoch < 90
>>> # ...
>>> scheduler = StepLR(optimizer, step_size=30, gamma=0.1)
>>> for epoch in range(100):
>>>     scheduler.step()  # is called at iteration start
>>>     train(...)
>>>     validate(...)
```

Example from `ignite.contrib` documentation ([doc link](https://pytorch.org/ignite/contrib/handlers.html#ignite.contrib.handlers.LRScheduler)):
```python
from ignite.contrib.handlers.param_scheduler import LRScheduler
from torch.optim.lr_scheduler import StepLR

step_scheduler = StepLR(optimizer, step_size=3, gamma=0.1)
scheduler = LRScheduler(step_scheduler)
trainer.add_event_handler(Events.ITERATION_COMPLETED, scheduler)  # Should this be Events.ITERATION_STARTED?
```

Minimum pytorch example to show behaviour of pytorch schedulers:
```python
import torch
optimizer = torch.optim.Adam([torch.Tensor([1,2,3])], lr=0.001) # initial learning rate is 0.001
lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer ,30)

lr_scheduler.get_lr() # outputs 0.01 instead of 0.001

lr_scheduler.step()
lr_scheduler.get_lr() # outputs 0.001 after first call of step()
```

### PR DESCRIPTION
**Pull request by Bibonaut**
Fixes #425 

Description:
According to [official pytorch documentation](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.StepLR), learning rate schedulers must be called at iteration start and not at iteration end:
```python
>>> # Assuming optimizer uses lr = 0.05 for all groups
>>> # lr = 0.05     if epoch < 30
>>> # lr = 0.005    if 30 <= epoch < 60
>>> # lr = 0.0005   if 60 <= epoch < 90
>>> # ...
>>> scheduler = StepLR(optimizer, step_size=30, gamma=0.1)
>>> for epoch in range(100):
>>>     scheduler.step()  # is called at iteration start
>>>     train(...)
>>>     validate(...)
```

Hence, events in doc examples from `ignite.contrib.handlers.param_scheduler`  are updated to use correct event `Events.ITERATION_STARTED` instead of `Events.ITERATION_COMPLETED`.

Check list:
* [ ] New tests are added (if a new feature is modified)
* [ ] New doc strings: text and/or example code are in RST format
* [x] Documentation is updated (if required)

### CODE REVIEW

**Comment by Bibonaut** [comment]
@vfdev-5 First try of my PR :) I read the linked information in your contribution guide but since this is my first PR on github (always used gitlab) and first open-source PR any advice is much appreciated!

**Comment by vfdev-5** [comment]
@Bibonaut thank you for the PR (or Merge Request if you prefer Gitlab terms :) )! 
Do not pay attention on failed CI it is broken atm.

**Comment by vfdev-5** [comment]
We'll merge it once CI become green and we could auto-build new docs

**Comment by vfdev-5** [comment]
@Bibonaut could you please merge the master into your branch

**Comment by vfdev-5** [comment]
@Bibonaut thanks !

**Comment by Bibonaut** [comment]
Thanks for finishing this PR :) I am on vacation atm. Hope, I can also contribute some code later when I am a bit more into ignite.
